### DISCUSSION TITLE
MaxMessageSize from 6.4.0 up to 7.0.0

### DISCUSSION THREAD

**Discussion by dank100** [rabbitmq/rabbitmq-stream-dotnet-client#283] [General]
From 6.4.0 and forward There is now a limit on the message size, as you rightfully explain in your changelogs:

"This release adds the ability to specify a maximum message size when receiving data. The default values are:

RabbitMQ .NET client 7.0.0 and beyond: 128MiB

RabbitMQ .NET client 6.4.0 up to 7.0.0: no limit by default"

However, when I specify the cf.MaxMessageSize = 536870912 as in your example. I'm still limited by the max size of the message, i.e. 134217728.

If I downgrade to 6.3.1, I'm not limited by the MessageSize, on the very same cluster and exchange.

I'm not sure this is any bug, but I cannot seem to figure it out.

An observation I've made in ConnectionFactory.cs is that on your main branch we have. 

          _public const uint DefaultMaxMessageSize = 134217728;_

 However, on, tag 6.5.0 we have.

      _public const uint DefaultMaxMessageSize = 0;_

In your docs you specify that 0 means "unlimited".

With me using 6.5.0, I would expect the MaxMessageSize being 0, whether I specify it or not. However, as explained I'm limited by the MaxMessageSize of 134217728.

Is this anything you've heard of before, and are you able to pinpoint me in the right direction?

- Dan

**Comment by Gsantomaggio** [comment]
You are in the wrong repository. Maybe you are looking for https://github.com/rabbitmq/rabbitmq-dotnet-client
